src/OrchardCore.Themes/TheAdmin/Assets/js/header/userPreferencesLoader.js

```diff
@@ -18,42 +18,46 @@ var observer = new MutationObserver(function (mutations) {
                     }
                     isCompactExplicit = adminPreferences.isCompactExplicit;
 
-                    if(adminPreferences.darkMode){
-                        document.getElementById('admin-darkmode').setAttribute('media', 'all');
-                        document.getElementById('admin-default').setAttribute('media', 'not all');
-                        body.classList.add('darkmode');
-                        
-                        document.getElementById('btn-darkmode').firstChild.classList.remove('fa-moon');
-                        document.getElementById('btn-darkmode').firstChild.classList.add('fa-sun');
-                    }
-                    else
-                    {
-                        document.getElementById('admin-darkmode').setAttribute('media', 'not all');
-                        document.getElementById('admin-default').setAttribute('media', 'all');
-                        body.classList.remove('darkmode');
+                    if (document.getElementById('admin-darkmode')) {
+                        if (adminPreferences.darkMode){
```

Comment:
Can you cache the selector in a var instead of calling getElementById many times ?